### PR TITLE
Revise examples in the intro to lists.

### PR DESCRIPTION
**Pull request by uzluisf**

### CODE REVIEW

File: doc/Type/List.pod6 [hunk 1]
```diff
@@ -24,31 +24,49 @@ information. The difference is that iteration generally treats a
...
     my @a = 1, 2, 3;
     for @a { }      # three iterations
-
-    my $s = @a;
-    for $s { }      # one iteration
     for @a.item { } # one iteration
-    for $s.list { } # three iterations
+
+C<.item> can often be written as C<$( ... )>, and on an array variable
+even as C<$@a>.
```
Comment (treyharris) [inline doc/Type/List.pod6]: This is a bit confusing to me. "Often"? When are `@a.item`, `$(@a)`, and `$@a` different and when are they the same?

Comment (jnthn) [reply doc/Type/List.pod6]: Both `$(@a)` and `$@a` are sugar for `@a.item`, so will behave the same. Maybe it was meaning it in the sense of "you'll often also see it written as"?

Comment (treyharris) [reply doc/Type/List.pod6]: @jnthn Got it, thank you.

@uzluisf May I suggest you either remove `often` or change `can often be written` to `is often written`?

Also, punctuation or lack thereof causes some ambiguity in the following clause. Please change:

`and on an array variable even as` →
- `and on array variables can even be written as` or 
- `and, on array variables, even as`?

Comment (uzluisf) [reply doc/Type/List.pod6]: @treyharris That sentence was already there :-). I just moved it up closer to where `.item` is first used in the code snippet. Is `C<.item> is often written as C<$( ... )>, and on array variables, even as C<$@a>.` fine enough?

BTW, the sentence at lines 23-25:

> The difference is that iteration generally treats a list (or any other list-like object, like a <Seq|/type/Seq> or an L<Array|/type/Array>) inside a scalar as a single element, **as long as it's part of another .**

seems incomplete. I have no idea to what "it's part of another" is referring to. Probably "it's part of another list."?

Comment (treyharris) [reply doc/Type/List.pod6]: > @treyharris That sentence was already there :-). I just moved it up closer to where `.item` is first used in the code snippet.

Yes, I noticed that, but unless you need to preserve the old wording for some reason (like you need Git to detect the movement), there's no reason _not_ to write it better while you're in there.

> Is `C<.item> is often written as C<$( ... )>, and on array variables, even as C<$@a>.` fine enough?

Almost, but `s:s/and on/and, on/`, please.

> BTW, the sentence at lines 23-25:
> 
> > The difference is that iteration generally treats a list (or any other list-like object, like a <Seq|/type/Seq> or an L<Array|/type/Array>) inside a scalar as a single element, **as long as it's part of another .**
> 
> seems incomplete. I have no idea to what "it's part of another" is referring to. Probably "it's part of another list."?

I don't understand this addition either... it looks like it was added (complete with the trailing space before the final full stop) by @moritz in 645466d:

```diff
@@ -18,7 +18,7 @@ Arrays to have every value of the list stored in a container.
 In Perl 6, assigning a C<List> to a scalar variable does not lose
 information. The difference is that iteration generally treats a
 list (or any other list-like object, like a L<Seq> or an L<Array>)
-inside a scalar as a single element.
+inside a scalar as a single element, as long as it's part of another .

     my @a = 1, 2, 3;                                                           
     for @a { }      # three iterations
```

Unless @moritz can illuminate, I'd say revert that line back to its `645466d~` version (in 
4801399f.

Comment (moritz) [reply doc/Type/List.pod6]: Sorry, I don't know anymore what I meant to write there. Feel free to discard it.

**Pull request merged**